components/zotero-service.js

```diff
@@ -612,42 +612,49 @@ ZoteroCommandLineHandler.prototype = {
 				}
 			}
 			if (param) {
-				addInitCallback(function (Zotero) {
+				addInitCallback(async function (Zotero) {
 					// Wait to handle things that require the UI until after it's loaded
-					Zotero.uiReadyPromise
-					.then(function () {
-						var file = Zotero.File.pathToFile(param);
-						
-						if(file.leafName.substr(-4).toLowerCase() === ".csl"
-								|| file.leafName.substr(-8).toLowerCase() === ".csl.txt") {
-							// Install CSL file
-							Zotero.Styles.install({ file: file.path }, file.path);
-						} else {
-							// Ask before importing
-							var checkState = {
-								value: Zotero.Prefs.get('import.createNewCollection.fromFileOpenHandler')
-							};
-							if (Components.classes["@mozilla.org/embedcomp/prompt-service;1"]
-									.getService(Components.interfaces.nsIPromptService)
-									.confirmCheck(null, Zotero.getString('ingester.importFile.title'),
-									Zotero.getString('ingester.importFile.text', [file.leafName]),
-									Zotero.getString('ingester.importFile.intoNewCollection'),
-									checkState)) {
-								Zotero.Prefs.set(
-									'import.createNewCollection.fromFileOpenHandler', checkState.value
-								);
-								
-								// Perform file import in front window
-								var wm = Components.classes["@mozilla.org/appshell/window-mediator;1"]
-								   .getService(Components.interfaces.nsIWindowMediator);
-								var browserWindow = wm.getMostRecentWindow("navigator:browser");
-								browserWindow.Zotero_File_Interface.importFile({
-									file,
-									createNewCollection: checkState.value
-								});
-							}
+					await Zotero.uiReadyPromise;
+					var file = Zotero.File.pathToFile(param);
+					
+					if (file.leafName.substr(-4).toLowerCase() === ".csl"
+							|| file.leafName.substr(-8).toLowerCase() === ".csl.txt") {
+						// Install CSL file
+						Zotero.Styles.install({ file: file.path }, file.path);
+					}
+					else if (file.leafName.substr(-4).toLowerCase() === ".pdf") {
+						let item = await Zotero.Attachments.importFromFile({
+							file,
+							libraryID: Zotero.Libraries.userLibraryID
```

Comment:
Should we have a dialog with a library drop-down, defaulting to the currently selected library? It's sort of weird if you're in another library and open a PDF via Zotero and then the PDF goes into your personal library.

Reply:
Arguably we should use that for metadata files too. If you only have a personal library, it would skip the prompt.

Reply:
Added a unified dialog. It looks like this:

<img width="363" alt="image" src="https://user-images.githubusercontent.com/1770299/156023137-1a869cd6-09ea-45d4-96f7-af864a2eb070.png">